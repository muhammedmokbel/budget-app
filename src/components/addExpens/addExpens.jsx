import React from 'react'
import ExpenseForm from '../ExpenseForm/ExpenseForm'

import {connect} from 'react-redux'
import {addExpense} from '../../redux/actionGenerators/expenseGenerators'

export class AddExpense extends React.Component {

  onSubmit = (expenseData) => {
    this.props.addexpense(expenseData)
    this.props.history.push('/')

  }


  render() {

    return (
      <ExpenseForm 
      onSubmit={this.onSubmit}/>
    )
  }

}


const mapDisptachToProps = dispatch => {
  return {
    addexpense : val => dispatch(addExpense(val))
  }
}

export default connect(undefined , mapDisptachToProps)(AddExpense);