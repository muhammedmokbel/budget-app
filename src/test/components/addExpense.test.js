import moment from 'moment'
import React from 'react' 
import { shallow } from 'enzyme'
import {AddExpense} from '../../components/addExpens/addExpens'


let submitSpy , history , wrapper; 

beforeEach(() => {
    submitSpy = jest.fn() 
    history = {push :jest.fn()  } 
    wrapper = shallow(<AddExpense addexpense={submitSpy} history={history} />)
})


test('should render addExpense component correctly' , () => {
    expect(wrapper).toMatchSnapshot()
})

test('should return submit data correctly' , () => {

    const dummyExpense = {
        title : 'test title', 
        note : 'test note', 
        amount : 3.1 , 
        createdAt : (moment(0))
    }

     wrapper.find('ExpenseForm').prop('onSubmit')(dummyExpense)

     expect(history.push).toHaveBeenCalledWith('/')
     expect(submitSpy).toHaveBeenCalledWith(dummyExpense)


})


