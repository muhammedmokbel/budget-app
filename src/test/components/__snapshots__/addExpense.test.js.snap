// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render addExpense component correctly 1`] = `
<ExpenseForm
  onSubmit={[Function]}
/>
`;
